src/c++/perf_analyzer/perf_analyzer.cc

```diff
@@ -428,13 +428,19 @@ PerfAnalyzer::WriteReport()
   bool should_output_metrics{
       params_->should_collect_metrics && params_->verbose_csv};
 
+  // (TMA-1526) Detect if the model is LLM and report LLM metrics based on that
+  // signal. Currently we just check if it's decoupled model.
+  bool should_output_llm_metrics{
+      parser_->IsDecoupled() && !params_->profile_export_file.empty()};
+
   std::unique_ptr<pa::ReportWriter> writer;
 
   FAIL_IF_ERR(
       pa::ReportWriter::Create(
           params_->filename, params_->targeting_concurrency(), perf_statuses_,
           params_->verbose_csv, profiler_->IncludeServerStats(),
-          params_->percentile, parser_, &writer, should_output_metrics),
+          params_->percentile, parser_, &writer, should_output_metrics,
+          collector_->GetData(), should_output_llm_metrics),
```

Comment:
does it make more sense to pass in the collector instead of the raw data? That would future proof us if more features are added to the collector, etc.###

Reply:
Agreed. Changed to passing the collector